### ISSUE TITLE
Support LazyProxy

### ISSUE THREAD

**Issue by azhe403** [enisn/DotNurseInjector#19]
Some problems like a Circular Dependency maybe solved by this lib.

https://github.com/cmendible/lazy-proxy-serviceprovider

Is possible support with this lib?

**Comment by enisn** [comment]
Yeah, a proxy can be injected instead of the original object and while calling a method on the interface, so the proxy can resolve that dependency and invoke the method.

In a short term, I can't say it'll be included or not. It's not the main goal of this library. But It would be awesome.

All PRs are welcome by the way 😉

**Comment by enisn** [comment]
I'll take a look at that feature.

**Comment by enisn** [comment]
I'm thinking about cleaner implementation. That library can be replaceable and it shouldn't be a dependency of the main project. I'm gonna create an extension named **DotNurse.Injector.LazyProxy**
